### DISCUSSION TITLE
Next step after building the database

### DISCUSSION THREAD

**Discussion by jsarriaa** [maize-genetics/phg_v2#164] [General]
Hi,

To set up a context, at our group we are working with barley assemblies building a database with PHG. At this point now we are able to imput fastq files, export its h.VCFs and get its fasta sequence file (only with full aligned chromosome sequences, we're still working on unplaced contigs to be included at the varieties genomes).

The next to scopes are:

- To somehow plot and represent the haplotypes and nodes of the graph from the hvcf, either the whole pangenome or only a specific region. Is it any tool that anyone would recommend us to try ourselves, if there is any?

-  Now, in order to bring to the next step and became useful the performed database, we would like to be able to align and search for 
small sequences, like genes. For now, we have experienced that if we provide a small fastq file, the output after find-paths still being a 
whole genome h.vcf file, supposedly finding the most likely to path. Actually, we would like to understand the criteria is followed to perform it. For instance, at the first 4 lines of one of our generated h.vcf files are:

```
##fileformat=VCFv4.2
##ALT=<ID=0000df22a39b9f4191853302d57f24e6,Description="haplotype data 
for line: 
Barke",Source="/genoma/nfs/nocontigs_phg//assemblies.agc",SampleName=Barke,Regions=chr7H_OX459908.1:13697931-13725513,Checksum=Md5,RefRange=chr7H_LR890102.1:13916748-13993213>
##ALT=<ID=0001ec3c5ff7c2a47a7b3d720def56b4,Description="haplotype data 
for line: 
HOR1168",Source="/genoma/nfs/nocontigs_phg//assemblies.agc",SampleName=HOR1168,Regions=chr1H_OX460229.1:450719287-450844842,Checksum=Md5,RefRange=chr1H_LR890096.1:448207310-448316135>
##ALT=<ID=0002058f0ef9c6c8eeebc60f31d67a79,Description="haplotype data 
for line: 
Barke",Source="/genoma/nfs/nocontigs_phg//assemblies.agc",SampleName=Barke,Regions=chr4H_OX459905.1:546215826-546303703,Checksum=Md5,RefRange=chr4H_LR890099.1:548784863-548855807>
##ALT=<ID=0002d86cf0d2381c76ebbe0ab7fa80f3,Description="haplotype data 
for line: 
HOR1168",Source="/genoma/nfs/nocontigs_phg//assemblies.agc",SampleName=HOR1168,Regions=chr5H_OX460233.1:583133310-583136540,Checksum=Md5,RefRange=chr5H_LR890100.1:578193865-578197081>
```

Each range belonging to different regions of different chromosomes. Is it any logical explanation to this sorting?
Coming back to the initial question, is it any way to get from the database the position of the sequence (which could be a fasta file) from 
the haplotypes, doing an alignment? We are planning to eventually provide this service to researchers/breeders from our pangenome in the future, It can be very useful.

Thank you in advance for your advices and considerations,
Joan

**Comment by lynnjo** [comment]
Hi Joan - To answer your question regarding  the sorting for the ##ALT headers:  these have been sorted based on the ID, which is why they appear random.  The ID field is the md5 hash value of the sequence, so not related to chrom or position.

We'll work on answers to your other questions.

**Reply by lynnjo** [reply]
Joan -  Initial response to your other questions:  

Plotting and nodes of the h.vcf:  We haven't yet had time to work on visualization though we agree it is important and we hope to have options in the future.

In the meantime, perhaps the rPHG2 functionality would be useful.  rPHG2 uses R methods to interact with phgv2 data.  There should be examples at the README from the web site that provide options for creating and visualizing matrix data from the phg:  https://github.com/maize-genetics/rPHG2

Another option might be importing your tiledb dataset to a notebook and running python tiledb queries.  There are various notebook examples on the web.  They access datasets from tiledb.cloud, but you wouldn't need that as you would use your own data.  Here is a link to one of their notebooks for an example.  You don't have to login to see the notebook, only to run it.
https://cloud.tiledb.com/notebooks/details/TileDB-Inc/tutorial_tiledbvcf_basics/preview

I'm not sure I understand your last question.  The sorting of the ALT headers, as I mentioned above, is based on the ID which is the md5 hash of the haplotype sequence.  For each haplotype, the ALT header has the sample name and coordinates that make up the sequence.  The ID in the ALT header should match the "ALT" parameters in the h.vcf body.  The "SampleName" and "Regions" list in the ALT header provide the  sample:chrom/coordinates that make up the sequence for that haplotype.
